libbeat/reader/readfile/timeout.go

```diff
@@ -67,10 +69,14 @@ func (r *TimeoutReader) Next() (reader.Message, error) {
 		r.running = true
 		go func() {
 			for {
-				message, err := r.reader.Next()
-				r.ch <- lineMessage{message, err}
-				if err != nil {
-					break
+				select {
+				case <-r.done:
+					return
+				case message, err := r.reader.Next():
+					r.ch <- lineMessage{message, err}
```

Comment:
we need the `r.ch <-` line to be in the select statement.

Reply:
:facepalm: